salt/utils/dns.py

```diff
@@ -1,18 +1,787 @@
+               for rr in resolver.query(name, rdtype, raise_on_no_answer=False)]
+        return res
+    except:
+        return False
+
+
+def _query_nslookup(name, rdtype, timeout=None, server=None):
+    '''
+    Use nslookup to lookup addresses
+    :param name: Name of record to search
+    :param rdtype: DNS record type
+    :param timeout: server response timeout
+    :param server: server to query
+    :return: [] of records or False if error
+    '''
+    cmd = 'nslookup -query={0} {1}'.format(rdtype, str(name))
+
+    if timeout is not None:
+        cmd += ' -timeout={0}'.format(int(timeout))
+    if server is not None:
+        cmd += ' {0}'.format(server)
+
+    cmd = __salt__['cmd.run_all'](cmd, python_shell=False, output_loglevel='quiet')
+    # In this case, 0 is not the same as False
+    if cmd['retcode'] != 0:
+        log.warning(
+            'nslookup returned ({0}): {1}'.format(
+                cmd['retcode'], cmd['stdout'].splitlines()[-1]
+            ))
+        return False
+
+    # ppr(cmd['stdout'].splitlines())
+
+    lookup_res = iter(cmd['stdout'].splitlines())
+
+    res = []
+    try:
+        line = ''
+        while True:
+            if name in line:
+                break
+            line = next(lookup_res)
+
+        while True:
+            # ppr('{}: now at {}'.format(name, line))
+            line = line.strip()
+            if not line or line.startswith('*'):
+                break
+            elif rdtype != 'CNAME' and 'canonical name' in line:
+                name = line.split()[-1][:-1]
+                line = next(lookup_res)
+                continue
+            elif rdtype == 'SOA':
+                line = line.split('=')
+            elif line.startswith('Name:'):
+                line = next(lookup_res)
+                line = line.split(':', 1)
+            elif line.startswith(name):
+                if '=' in line:
+                    line = line.split('=', 1)
+                else:
+                    line = line.split(' ')
+
+            res.append(line[-1].strip(string.whitespace + '"'))
+            line = next(lookup_res)
+
+    except StopIteration:
+        pass
+    finally:
+        if rdtype == 'SOA':
+            return [' '.join(res[1:])]
+        else:
+            return res
+
+
+def query(
+        name,
+        rdtype,
+        method=None,
+        servers=None,
+        timeout=None,
+        secure=None
+):
+    '''
+    Lookup DNS records
+    :param name: name to lookup
+    :param rdtype: DNS record type
+    :param method: simple, pydns, dig, drill, host, nslookup or auto (default)
+    :param servers: (list of) server(s) to try in-order
+    :param timeout: query timeout or a valiant approximation of that
+    :param secure: return only DNSSEC secured responses
+    :return: [] of record data
+    '''
+    # opts = __opts__.get('dns', {})
+    opts = {}
+    rdtype = rdtype.upper()
+
+    query_methods = (
```

Comment:
The linter does not like the extra whitespace after the commas, please remove it.